### ISSUE TITLE
Make `show` callback accept futures

### ISSUE THREAD

**Issue by ueman** [ueman/feedback#96]
It the `show` callback accepts future, this lib could also present a "Thank you screen". It's also probably a better ergonomic because most user are doing async work in it anyway.

**Comment by seanradford** [comment]
With this enhancement, having the ability to show a progress indicator whilst the future has not completed would be nice.

**Comment by ueman** [comment]
Yep, exactly. That would allow us to show a loading indicator to the user while the feedback is uploading.
